src/rebar_dir.erl

```diff
@@ -93,8 +93,14 @@ global_config() ->
     filename:join([Home, ".config", "rebar3", "rebar.config"]).
 
 global_cache_dir(Opts) ->
-    Home = home_dir(),
-    rebar_opts:get(Opts, global_rebar_dir, filename:join([Home, ".cache", "rebar3"])).
+    RebarCacheDir = case os:getenv("REBAR_CACHE_DIR") of
+        false ->
+            Home = home_dir(),
+            filename:join([Home, ".cache", "rebar3"]);
+        ConfigFile ->
+            ConfigFile
+    end,
+    rebar_opts:get(Opts, global_rebar_dir, RebarCacheDir).
```

Comment:
The logic is fine here, I'm just wondering if we have qualms about going to the OS and read the ENV on every single call of this function.

Feels like we should do it once in `rebar3.erl` or in core, and then reuse it after that. @tsloughter any opinions?